csharp/src/Ice/Communicator.cs

```diff
@@ -1,297 +1,386 @@
+        return reference is not null ? new ObjectPrxHelper(reference) : null;
+    }
 
     /// <summary>
-    /// Convert a proxy into a string.
+    /// Converts a proxy into a string.
     /// </summary>
-    /// <param name="obj">The proxy to convert into a stringified proxy.
-    ///  </param>
-    /// <returns>The stringified proxy, or an empty string if
-    ///  obj is nil.
-    ///  </returns>
-    string proxyToString(ObjectPrx obj);
+    /// <param name="proxy">The proxy to convert into a stringified proxy.</param>
+    /// <returns>The stringified proxy, or an empty string if <paramref name="proxy" /> is null.</returns>
+    public string proxyToString(ObjectPrx? proxy) =>
+        proxy is null ? "" : ((ObjectPrxHelperBase)proxy).iceReference().ToString();
 
     /// <summary>
-    /// Convert a set of proxy properties into a proxy.
-    /// The "base" name supplied in the property argument
-    ///  refers to a property containing a stringified proxy, such as MyProxy=id:tcp -h localhost -p 10000.
-    ///  Additional properties configure local settings for the proxy, such as MyProxy.PreferSecure=1. The
-    ///  "Properties" appendix in the Ice manual describes each of the supported proxy properties.
+    /// Converts a set of proxy properties into a proxy. The "base" name supplied in the property argument refers to a
+    /// property containing a stringified proxy, such as "MyProxy=id:tcp -h localhost -p 10000". Additional properties
+    /// configure local settings for the proxy, such as MyProxy.PreferSecure=1. The "Properties" appendix in the Ice
+    /// manual describes each of the supported proxy properties.
     /// </summary>
-    ///  <param name="property">The base property name.
-    ///  </param>
+    ///  <param name="property">The base property name.</param>
     /// <returns>The proxy.</returns>
-    ObjectPrx propertyToProxy(string property);
+    public ObjectPrx? propertyToProxy(string property)
+    {
+        string proxy = instance.initializationData().properties.getProperty(property);
+        Reference? reference = instance.referenceFactory().create(proxy, property);
+        return reference is not null ? new ObjectPrxHelper(reference) : null;
+    }
 
     /// <summary>
-    /// Convert a proxy to a set of proxy properties.
+    /// Converts a proxy to a set of proxy properties.
     /// </summary>
-    /// <param name="proxy">The proxy.
-    ///  </param>
-    /// <param name="property">The base property name.
-    ///  </param>
+    /// <param name="proxy">The proxy.</param>
+    /// <param name="prefix">The base property name.</param>
     /// <returns>The property set.</returns>
-    Dictionary<string, string> proxyToProperty(ObjectPrx proxy, string property);
+    public Dictionary<string, string> proxyToProperty(ObjectPrx proxy, string prefix) =>
+        proxy is null ? [] : ((ObjectPrxHelperBase)proxy).iceReference().toProperty(prefix);
```

Comment:
We usually don't have null checks, for nonnullable parameters.

Reply:
You're right, remove this check.